File: src/main/scala/com/martomate/tripaint/model/image/format/StorageFormat.scala
package com.martomate.tripaint.model.image.format

import com.martomate.tripaint.model.coords.{StorageCoords, TriangleCoords}

trait StorageFormat {
  def transform(coords: TriangleCoords): StorageCoords

  def reverse(coords: StorageCoords): TriangleCoords
}

File: core/src/main/kotlin/tripaint/image/format/StorageFormat.kt
package tripaint.image.format

import tripaint.coords.StorageCoords
import tripaint.coords.TriangleCoords

interface StorageFormat {
    fun supportsImageSize(size: Int): Boolean

    fun transform(coords: TriangleCoords): StorageCoords

    fun reverse(coords: StorageCoords): TriangleCoords
}
